gcp-client/src/main/java/com/google/graphite/platforms/plugin/client/ComputeClient.java

```diff
@@ -0,0 +1,541 @@
+    List<Image> images = compute.images().list(projectId).execute().getItems();
+    if (images == null) {
+      return new ArrayList<>();
+    }
+
+    // No deprecated items
+    Iterator it = images.iterator();
+    while (it.hasNext()) {
+      Image o = (Image) it.next();
+      if (o.getDeprecated() != null
+          && o.getDeprecated().getState().equalsIgnoreCase("DEPRECATED")) {
+        it.remove();
+      }
+    }
+
+    images.sort(Comparator.comparing(Image::getName));
+
+    return images;
+  }
+
+  public Image getImage(String projectId, String name) throws IOException {
+    Image image = compute.images().get(projectId, name).execute();
+
+    return image;
+  }
+
+  public List<AcceleratorType> getAcceleratorTypes(String projectId, String zone)
+      throws IOException {
+    zone = nameFromSelfLink(zone);
+
+    List<AcceleratorType> acceleratorTypes =
+        compute.acceleratorTypes().list(projectId, zone).execute().getItems();
+    if (acceleratorTypes == null) {
+      return new ArrayList<>();
+    }
+
+    // No deprecated items
+    Iterator it = acceleratorTypes.iterator();
+    while (it.hasNext()) {
+      AcceleratorType o = (AcceleratorType) it.next();
+      if (o.getDeprecated() != null
+          && o.getDeprecated().getState().equalsIgnoreCase("DEPRECATED")) {
+        it.remove();
+      }
+    }
+    acceleratorTypes.sort(Comparator.comparing(AcceleratorType::getName));
+    return acceleratorTypes;
+  }
+
+  public List<Network> getNetworks(String projectId) throws IOException {
+    List<Network> networks = compute.networks().list(projectId).execute().getItems();
+
+    if (networks == null) {
+      return new ArrayList<>();
+    }
+    return networks;
+  }
+
+  public List<Subnetwork> getSubnetworks(String projectId, String networkSelfLink, String region)
+      throws IOException {
+    region = nameFromSelfLink(region);
+    List<Subnetwork> subnetworks =
+        compute.subnetworks().list(projectId, region).execute().getItems();
+    if (subnetworks == null) {
+      return new ArrayList<>();
+    }
+
+    // Only subnetworks in the parent network
+    Iterator it = subnetworks.iterator();
+    while (it.hasNext()) {
+      Subnetwork o = (Subnetwork) it.next();
+      if (!o.getNetwork().equals(networkSelfLink)) {
+        it.remove();
+      }
+    }
+
+    subnetworks.sort(Comparator.comparing(Subnetwork::getName));
+    return subnetworks;
+  }
+
+  public Operation insertInstance(String projectId, String template, Instance instance)
+      throws IOException {
+    final Compute.Instances.Insert insert =
+        compute.instances().insert(projectId, instance.getZone(), instance);
+    if (!Strings.isNullOrEmpty(template)) {
+      insert.setSourceInstanceTemplate(template);
+    }
+    return insert.execute();
+  }
+
+  public Operation terminateInstance(String projectId, String zone, String InstanceId)
+      throws IOException {
+    zone = nameFromSelfLink(zone);
```

Comment:
Prefer not changing the values of function params. Consider setting them to `final`

Reply:
Done